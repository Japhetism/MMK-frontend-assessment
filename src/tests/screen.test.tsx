import React from 'react';
import "@testing-library/jest-dom";
import { shallow } from 'enzyme';

import Screen from '../screens';
import assert from 'assert';

describe("The component rendered", () => {

  const wrapper = shallow(<Screen />);

  it('renders the Screen component without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain one timer component', () => {
    assert.equal(wrapper.find('Timer').length === 1, true);
  });

  it('should contain one typing component', () => {
    assert.equal(wrapper.find('Typing').length === 1, true);
  })
});