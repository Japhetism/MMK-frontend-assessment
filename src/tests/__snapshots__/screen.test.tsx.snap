// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component rendered renders the Screen component 1`] = `
<styled.div>
  <styled.div>
    <Timer
      durationOptions={
        Array [
          1,
          2,
          3,
          4,
          5,
        ]
      }
      setDisableUserInput={[Function]}
      setShowResult={[Function]}
    />
  </styled.div>
  <styled.div>
    <Typing
      disableUserInput={true}
      showResult={false}
    />
  </styled.div>
</styled.div>
`;

exports[`The component rendered renders the Screen component without crashing 1`] = `
<styled.div>
  <styled.div>
    <Timer
      durationOptions={
        Array [
          1,
          2,
          3,
          4,
          5,
        ]
      }
      setDisableUserInput={[Function]}
      setShowResult={[Function]}
    />
  </styled.div>
  <styled.div>
    <Typing
      disableUserInput={true}
      showResult={false}
    />
  </styled.div>
</styled.div>
`;

exports[`The component rendered renders the screen component 1`] = `
<styled.div>
  <styled.div>
    <Timer
      durationOptions={
        Array [
          1,
          2,
          3,
          4,
          5,
        ]
      }
      setDisableUserInput={[Function]}
      setShowResult={[Function]}
    />
  </styled.div>
  <styled.div>
    <Typing
      disableUserInput={true}
      showResult={false}
    />
  </styled.div>
</styled.div>
`;

exports[`renders the screen component 1`] = `
<styled.div>
  <styled.div>
    <Timer
      durationOptions={
        Array [
          1,
          2,
          3,
          4,
          5,
        ]
      }
      setDisableUserInput={[Function]}
      setShowResult={[Function]}
    />
  </styled.div>
  <styled.div>
    <Typing
      disableUserInput={true}
      showResult={false}
    />
  </styled.div>
</styled.div>
`;
